.github/workflows/publish.yml

```diff
@@ -0,0 +1,64 @@
+name: Publish to PyPI [Production]
+on:
+  release:
+    types: [published]
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up python
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      #----------------------------------------------
+      #  -----  install & configure poetry  -----
+      #----------------------------------------------
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+      #----------------------------------------------
+      #       load cached venv if cache exists
+      #----------------------------------------------
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ github.event.repository.name }}-${{ hashFiles('**/poetry.lock') }}
+      #----------------------------------------------
+      # install dependencies if cache does not exist
+      #----------------------------------------------
+      - name: Install dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-root
+      #------------------------------------------------------------------------------------------------
+      # Here we use version-increment to fetch the latest tagged version (we won't increment it though)
+      #------------------------------------------------------------------------------------------------
+      - name: Get next version
+        uses: reecetech/version-increment@2022.2.4
```

Comment:
we have this in the other publish yaml file to publish to test pypi. 

How do we handle this scenario:
1. we publish to test pypi, hence publish test action increments the version
2. we want to publish the same binary in step 1 to pypi (do we have to roll back the version manually after step 1. incremented it?)

Reply:
Pypi does not allow rollbacks. test-pypi is entirely different from Pypi. Pushing a version to one has zero affect on the other.

Note that the output from this step is never used as described in the comments. 

The "one true version" is derived from git tags. We're using the `reecetech/version-increment` action as a simple shortcut to parse that the version tags.